### ISSUE TITLE
Remove the absolutePath property from the IFluidHandle interface

### ISSUE THREAD

**Issue by sonalivdeshpande** [microsoft/FluidFramework#8271]
Remove the `absolutePath` property from the `IFluidHandle` interface (`\common\lib\core-interfaces\src\handles.ts`). We want to discourage the use of absolute path; therefore, this property should be removed. To replace its functionality, the `get` function can obtain the underlying object to access the absolute path. All references to this property must first be removed before deleting.

Work items that have to be resolved before we can remove absolutePath:
- [ ] https://github.com/microsoft/FluidFramework/issues/3953
- [ ] https://github.com/microsoft/FluidFramework/issues/9235

**Comment by ssimic2** [comment]
@sdeshpande3 I'm not sure I am clear on the above comment: `To replace its functionality, the get function can obtain the underlying object to access the absolute path`.

Original comment was `@deprecated - Do not use handle's path for routing. Use `get` to get the underlying object.`. Is the desire here just to discourage use of IDs and move to handles? If that is the case, we already have ticket for that: https://github.com/microsoft/FluidFramework/issues/8283.

It seems that `absolutePath` still provides value. Ex. Serializer uses it to serialize handle.

**Comment by ssimic2** [comment]
Hey @sdeshpande3 could you comment on this one so we can move it forward. We chatted about it, just want to make sure we collect all the feedback here.

**Comment by sdeshpande3** [comment]
@scottn12, did you write down this description?

**Comment by scottn12** [comment]
> @scottn12, did you write down this description?

Yes, I did. I Just summarized what I found looking at the message associated with the `@deprecated` tag. I believe the idea is to discourage using the `absolutePath` to access the underlying fluid object, and instead use `get()` for the same purpose.

**Comment by sdeshpande3** [comment]
FYI: @ChumpChief

**Comment by ssimic2** [comment]
@ChumpChief let us know your thoughts here.

It seems it make more sense to leave absolutePath in place.
1. We are tackling the issue Scott mentioned, here: https://github.com/microsoft/FluidFramework/issues/8283. It seems a separate issue.
2. `absolutePath` seems used for other purposes, not just to get the underlying fluid object.

**Comment by agarwal-navin** [comment]
We want to remove `absolutePath` because it is being used to route to objects within a container and we want to remove that path.
Basically, the handle's `absolutePath` is stored in a DDS and is later used to get the underlying object by calling `request` on the path. All internal routing to objects within a container should be done via storing / passing handles and calling `get` to the underlying object.
However, we haven't completely solved all the scenarios associated with it so we cannot move to solely using handles for routing. So, we cannot get rid of `absolutePath` for now. Also, as Sasha mentioned, we are using it internally to serialize handles and for GC.

@vladsud might have more thoughts on what the path forward should be.

**Comment by ssimic2** [comment]
Yes, it seems like RemoteFluidObjectHandle is also using this field to complete `get` calls. If this field is derived out of serialized (existing snapshot) data, I'm not sure how we can remove it and still ensure data persistence compatibility. I could be wrong here.

I'll add it to "Needs Solutioning" for now.

**Comment by ssimic2** [comment]
Let's track all the "Persisted data compatibility" tickets here: https://github.com/microsoft/FluidFramework/issues/8774
